kernel/tests/golden_tables.rs

```diff
@@ -0,0 +1,343 @@
+//! Golden table tests adapted from delta-io/delta tests.
+//!
+//! Data (golden tables) are stored in tests/golden_data/<table_name>.tar.zst
+//! Each table directory has a table/ and expected/ subdirectory with the input/output respectively
+
+use arrow::{compute::filter_record_batch, record_batch::RecordBatch};
+use arrow_array::ArrayRef;
+use arrow_select::concat::concat_batches;
+use paste::paste;
+use std::path::Path;
+use std::sync::Arc;
+
+use arrow_schema::SchemaRef as ArrowSchemaRef;
+use delta_kernel::{engine::arrow_data::ArrowEngineData, DeltaResult, Table};
+use futures::{stream::TryStreamExt, StreamExt};
+use object_store::{local::LocalFileSystem, ObjectStore};
+use parquet::arrow::async_reader::{ParquetObjectReader, ParquetRecordBatchStreamBuilder};
+
+use arrow_array::Array;
+use arrow_schema::DataType;
+use delta_kernel::engine::default::executor::tokio::TokioBackgroundExecutor;
+use delta_kernel::engine::default::DefaultEngine;
+
+#[macro_use]
+mod common;
+use common::to_arrow;
+
+// load the test data from <test_table>.tar.zst file
+fn load_test_data(test_name: &str) -> Result<tempfile::TempDir, Box<dyn std::error::Error>> {
+    let path = format!("tests/golden_data/{}.tar.zst", test_name);
+    let tar = zstd::Decoder::new(std::fs::File::open(path)?)?;
+    let mut archive = tar::Archive::new(tar);
+    let temp_dir = tempfile::tempdir()?;
+    archive.unpack(temp_dir.path())?;
+    Ok(temp_dir)
+}
+
+// NB adapated from DAT: read all parquet files in the directory and concatenate them
+async fn read_expected(path: &Path) -> DeltaResult<Option<RecordBatch>> {
```

Comment:
There is no reason for this to return a result of an option (we never return Ok(None)).

so just:

```rust
async fn read_expected(path: &Path) -> DeltaResult<Option<RecordBatch>> {
   ...
   Ok(all_data)
}
```
and then no need to unwrap the option below.

This is busted in DAT too, so maybe fix it up there.

Reply:
ah yep artifact of DAT copypasta :)